kit/cli/viper.go

```diff
@@ -137,8 +121,15 @@ func BindOptions(v *viper.Viper, cmd *cobra.Command, opts []Opt) {
 			} else {
 				flagset.StringVar(destP, o.Flag, d, o.Desc)
 			}
-			mustBindPFlag(v, o.Flag, flagset)
-			*destP = v.GetString(envVar)
+			if err := v.BindPFlag(o.Flag, flagset.Lookup(o.Flag)); err != nil {
+				return err
+			}
+			if envVal != nil {
+				if s, err := cast.ToStringE(envVal); err == nil {
```

Comment:
This implementation (and the repetitious blocks below) ignores any errors returned when trying to convert a value pulled from config / the environment into the type expected by the CLI flags. It's a faithful translation of the previous behavior (see the impl for `v.GetString` etc), but I'm surprised it hasn't caused any user frustration. IMO it'd be better to fail fast here when an invalid value is provided in config / the environment, or at least to print a warning. Thoughts?

Reply:
Yes, failing fast on invalid config is the right behavior.

Reply:
Silently ignoring config flags definitely sounds bad to me.